openstack/orchestration/v1/stacks/utils.go

```diff
@@ -116,7 +119,7 @@ func (t *TE) Parse() error {
 			return ErrInvalidDataFormat{}
 		}
 	}
-	return t.Validate()
+	return t.Validate() // FIXME This always calls TE.Validate, not Template.Validate/Environment.Validate
```

Comment:
I think the easiest way to deal with this would be to change this line here to `return nil` and then do something like:

```go
if err := childTemplate.Parse(); err == nil {
  if err := childTemplate.Validate(); err == nil {
    if err := childTemplate.getFileContents(childTemplate.Parsed, ignoreIf, recurse); err != nil {
      return err
    }
  }
}
```

Reply:
Ok, in that case, shall I also remove ```func (t *TE) Validate() error```?

Reply:
Yes, let's go ahead and remove it.

Reply:
Ok, done!